components/modal/addCollectionModal.tsx

```diff
@@ -48,7 +47,11 @@ export default function AddCollectionModal({ closeModal, collection, isOpen }: A
     }).catch(err => {
       console.error(err);
       toast.dismiss();
-      toast.error('Error adding collection');
+      toast.error('Error: Name is longer than the maximum allowed length (50 characters)',
```

Comment:
Instead of writing the error message here, can you use the `err` object to get the message given by the server? If you make it async (`async err => ...`), then you can do `JSON.parse(await err)?.error` to get the error message